public/javascript/scripts.js

```diff
@@ -0,0 +1,278 @@
+
+var globalInput = "";                   //input string will store the input typed in a keyboard up until the user presses enter
+var selectedInput = "none";             //currently selected input on the assentInfo table. e.x. 'asset_type' OR 'customer'
+var selectedTableAsset = "none";        //currently selected asset from the table. Will be set as the tag of the asset selected from the table. ex. 123456A
+
+var currentAsset = {};
+
+const MACRO_LETTERS = "ABYZ";            //Contains all of the recognized macro letter endings 
+const MACRO_LOAD_ASSET_LETTERS = "ABZ";  //Contains macro letters that will load an asset or create a new one if it is not in the db
+
+//variables that will hold HTML for different elements in the 
+var defaultAssetRow = "";
+var assetInfoNoChange = "";
+var assetInfoText = "";
+var assetInfoDate = "";
+var assetInfoNumber = "";
+var assetInfoSelect = "";
+
+$(document).ready(function() {
+    $(".signInBadge").focus();
+    
+    $(".signInBadge").keypress(function(e){
+        if(e.key == "Enter") {
+            if($(".signInBadge").val() == "123456Y") {
+                $(".backgroundCover").hide();
+                $(".signInContainer").hide();
+                initialize();
+                
+                //instead of setting to the badge here, set it to the username received from the server
+                $("#user").html($(".signInBadge").val());
+            }
+            $(".signInBadge").val('');
+        }
+    });
+    
+    $("body").keypress(function(e){
+        if(selectedInput == "none") {
+            if(e.key == "Enter") {
+                processInput(globalInput);
+                globalInput = "";
+            }
+            else {
+                globalInput += e.key;
+            }
+        }
+    });
+});
+
+function initialize() {
+    //load the data from the server here josh
+    var fd = [];
+    for(var i = 0; i < 30; i ++) {
+        fd[i] = fakeData;
+    }
```

Comment:
This 'fd' array is the array of fake data, used just for testing, which is why it has such a lame name... it will be replaced by an api call to the server.